tarantool_test.go

```diff
@@ -1005,3 +1006,119 @@ func TestComplexStructs(t *testing.T) {
 		return
 	}
 }
+
+
+var uuidSpace = "testUUID"
+var uuidIndex = "primary"
+
+type TupleUUID struct {
+	id uuid.UUID
+}
+
+func (t *TupleUUID) DecodeMsgpack(d *msgpack.Decoder) error {
+	var err error
+	var l int
+	if l, err = d.DecodeSliceLen(); err != nil {
+		return err
+	}
+	if l != 1 {
+		return fmt.Errorf("array len doesn't match: %d", l)
+	}
+	res, err := d.DecodeInterface()
+	if err != nil {
+		return err
+	}
+	t.id = res.(uuid.UUID)
+	return nil
+}
+
+func connectWithValidation(t *testing.T) *Connection {
+	conn, err := Connect(server, opts)
+	if err != nil {
+		t.Errorf("Failed to connect: %s", err.Error())
+	}
+	if conn == nil {
+		t.Errorf("conn is nil after Connect")
+	}
+	return conn
+}
+
+func skipIfUUIDUnsupported(t *testing.T, conn *Connection) {
+	resp, err := conn.Eval("return pcall(require('msgpack').encode, require('uuid').new())", []interface{}{})
+	if err != nil {
+		t.Errorf("Failed to Eval: %s", err.Error())
+	}
+	if resp == nil {
+		t.Errorf("Response is nil after Eval")
+	}
+	if len(resp.Data) < 1 {
+		t.Errorf("Response.Data is empty after Eval")
+	}
+	val := resp.Data[0].(bool)
+	if val != true {
+		t.Skip("Skipping test for Tarantool without UUID support in msgpack")
+	}
+}
+
+func TestUUIDselect(t *testing.T) {
+	conn := connectWithValidation(t)
+	defer conn.Close()
+
+	skipIfUUIDUnsupported(t, conn)
+
+	id, uuidErr := uuid.Parse("c8f0fa1f-da29-438c-a040-393f1126ad39")
+	if uuidErr != nil {
+		t.Errorf("Failed to prepare test uuid: %s", uuidErr)
+	}
+
+	resp, errSel := conn.Select(uuidSpace, uuidIndex, 0, 1, IterEq, []interface{}{ id })
+	if errSel != nil {
+		t.Errorf("UUID select failed: %s", errSel.Error())
+	}
+	if resp == nil {
+		t.Errorf("Response is nil after Select")
+	}
+	if len(resp.Data) != 1 {
+		t.Errorf("Response Data len != 1")
+	}
```

Comment:
We don't check data in the response of `Select()` except for length of the tuple. Is it intentional?

Reply:
Added checks for tuples data in TestSelect and TestReplace cases